src/ngx_http_lua_directive.c

```diff
@@ -464,8 +466,20 @@ ngx_http_lua_rewrite_by_lua(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
         return NGX_CONF_ERROR;
     }
 
-    if (llcf->rewrite_handler) {
-        return "is duplicate";
+    if (p_rewrites == NULL) {
+        p_rewrites = ngx_array_create(cf->pool, 8,
+                                      sizeof(ngx_http_lua_rewrites_t));
+
+        if (p_rewrites == NULL) {
+            return "not enough memory";
```

Comment:
The common practice is to simply return `NGX_CONF_ERROR` in this case. See other similar places in this C function.